test/topology/test_alter_delete_regression.py

```diff
@@ -0,0 +1,57 @@
+#
+# Copyright (C) 2022-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+"""This is a regression test for #6174"""
+
+
+import asyncio
+import logging
+from test.pylib.random_tables import Column, IntType, BigIntType
+import pytest
+
+
+logger = logging.getLogger("test_alter_delete")
+
+
+LOOPS          =     4    # Attempts to trigger failure
+NROWS          =  1000    # Total rows on table
+MAX_S          =    10.0  # Max run seconds
+ALTER_TABLE_S  =     0.1  # Alter table at 100ms
+
+
+@pytest.mark.asyncio
+async def test_new_table(manager, random_tables):
+    """Test LWT regression causing hangs when doing DELETEs while there's an ALTER TABLE"""
+    cql = manager.cql
+    assert cql is not None
+
+    # Prologue
+    table = await random_tables.add_table(columns=[Column('pk', IntType),
+                                                   Column('v', IntType),
+                                                   Column('d1', IntType),
+                                                   Column('d2', IntType)])
+    await asyncio.gather(*(table.insert_seq() for _ in range(NROWS)))
+    logger.debug("Done prologue")
+
+    async def delete_rows():
+        logger.debug("delete_rows: start")
+        for i in range(0, NROWS - 1):
+            table.delete_pk(i)
+            await asyncio.sleep(.001)
+        logger.debug("delete_rows: done")
+
+    async def alter_table():
+        await asyncio.sleep(ALTER_TABLE_S)
+        logger.debug("alter_table: start")
+        await table.drop_column("d2")
+        await table.add_column(Column("d2", BigIntType))
```

Comment:
Is concurrency, and 1000 rows and doing 1000 deletions in parallel with the alter table, really necessary for this test? 
What I mean is, can't the same thing be reproduced with writing just two rows, deleting one of them, then altering the table and then trying to delete the second row with the existing connection?

The reason I'm asking this isn't just to simplify this test, but also to make it less dependent on timing issues. And also take less time to run.

Reply:
Back when I made the dtest repro this was the smallest I could do to trigger it.

This runs in 3 seconds in my local machine.

Reply:
Ok, I was just worried I don't understand what this bug was about. I didn't realize it was about concurrency. I thought it was about prepared statements being cached beyond their usefulness.

What surprised me was that this test only runs, as far as I can tell, the ALTER TABLE operation once (actually two commands each doing half of the change). So if it really relies on concurrency, how often does this single ALTER TABLE happen to coincide with a delete? Does this test fail (before being fixed) always? Frequently? Sometimes?
I hope you did try to run this test on the old code and see it reproduces the bug - right?

Reply:
> I hope you did try to run this test on the old code and see it reproduces the bug - right?

You got me. I couldn't repro because I had trouble building 2020 scylla and also was having problems with new dtest.

I'll pick this later as it's taking way too much time.